File: R functions/Cmatrixmultip.R
Cmatrixmultip=function(c,vec,Ways){
  if (length(c)!= length(vec)){
    stop('length c is different  from  length vec, please insert correct values')
  }
  c_hat=fft(c, inverse = FALSE)
  vec_hat=fft(vec,inverse = FALSE)
  switch(Ways,
         reg={
  prod=vector('double',length(c))
  prod=vec_hat*c_hat
  C=vector('double',length(c))
  C=fft(prod,inverse = TRUE)*(1/length(c))
  return(C)
  },
  inv={
  div=vector('double',length(c))
  div=vec_hat/c_hat
  C=vector('double',length(c))
  C=fft(div,inverse = TRUE)*(1/length(c))
  return(C)
  },
  stop('Ways can be either reg or inv'))
}

File: Matlab Functions/Cmatrixmultip.m
function C = Cmatrixmultip(c,vec,varargin)
warning('c must be the first column  of the Circular Matrix')
if length(c)~= length(vec)
     error('length c  (%i) is different  from  length vec (%i), please insert correct values',length(c),length(vec))
end
defaultMultiplication = 'reg';
expectedMultiplication = {'reg','inv'};

p=inputParser;
p.CaseSensitive = false;
validScalarPosNum = @(x)  isnumeric(x);
addRequired(p,'c',validScalarPosNum);
addRequired(p,'vec',validScalarPosNum);
addParameter(p,'Ways',defaultMultiplication, @(x) any(validatestring(x,expectedMultiplication)));
parse(p,c,vec,varargin{:});

if strcmp(p.Results.Ways, 'reg')
  C = ifft(fft(c).*fft(vec));
end

if strcmp(p.Results.Ways, 'inv')
  C = ifft( fft(vec)./fft(c));
end

end

